src/util/kafka/index.js

```diff
@@ -272,6 +272,36 @@ const commitMessageSync = async (kafkaTopic, consumer, message) => {
   }
 }
 
+const proceed = async (defaultKafkaConfig, params, opts) => {
+  const { message, kafkaTopic, consumer } = params
+  const { consumerCommit, fspiopError, producer, fromSwitch, toDestination } = opts
+  let metadataState
+
+  if (consumerCommit) {
+    await commitMessageSync(kafkaTopic, consumer, message)
+  }
+  if (fspiopError) {
+    if (!message.value.content.uriParams || !message.value.content.uriParams.id) {
+      message.value.content.uriParams = { id: decodePayload(params.message.value.content.payload).transferId }
```

Comment:
I am not sure this should be done here. The reason is that it will be an expensive operation since we should already be decoding the message when receiving it by the consumer. I think we need to discuss this going forward.

Reply:
Ok removed